### PR TITLE
Hungermod Improve

### PR DESCRIPTION
**Pull request by 00875**
Added Config.hungertimer and Config.maxhunger and Config.hungerexcludes
Fixed missing sound
HMThink does not need to run on every frame (Think), so running it on timer
Code fixed up

### CODE REVIEW

File: entities/entities/food/init.lua [hunk 1]
```diff
@@ -25,19 +25,27 @@ function ENT:OnTakeDamage(dmg)
...
 		effectdata:SetRadius(3)
 		util.Effect("Sparks", effectdata)
-		self:Remove()
+		SafeRemoveEntity(self)
 	end
 end
 
 function ENT:Use(activator,caller)
-	caller:setSelfDarkRPVar("Energy", math.Clamp((caller:getDarkRPVar("Energy") or 0) + 100, 0, 100))
+	local override = self.foodItem and self.foodItem.onEaten and self.foodItem.onEaten(self, activator, self.foodItem) or isfunction(self.onEaten) and self.onEaten(self, activator, self.foodItem)
```
Comment (FPtje) [inline entities/entities/food/init.lua]: And where would food get the "foodItem" variable from?

File: entities/entities/food/init.lua [hunk 2]
```diff
@@ -25,19 +25,27 @@ function ENT:OnTakeDamage(dmg)
...
+
+	if override then
+		SafeRemoveEntity(self)
+		return
+	end
+	
+	caller:setSelfDarkRPVar("Energy", GAMEMODE.Config.maxhunger)
```
Comment (FPtje) [inline entities/entities/food/init.lua]: Why should the "max hunger" ever by anything other than 100? You already have hungerspeed.

File: gamemode/modules/hungermod/cl_init.lua [hunk 3]
```diff
@@ -82,7 +82,7 @@ local function HMHUD()
 
 	if FoodAteAlpha > -1 then
 		local mul = 1
-		if FoodAteY <= ScrH() - 100 then
+		if FoodAteY <= ScrH() - 100 then --??? What is dis, Falco?
```
Comment (FPtje) [inline gamemode/modules/hungermod/cl_init.lua]: Immature to add questions in god damn comments in code.

File: gamemode/modules/hungermod/sv_hungermod.lua [hunk 4]
```diff
@@ -1,27 +1,34 @@
...
 local function HMThink()
-	if not GAMEMODE.Config.hungerspeed then return end
+	if !GAMEMODE.Config.hungerspeed then return end
+	local time = GAMEMODE.Config.hungertimer or 10
 
 	for k, v in pairs(player.GetAll()) do
-		if v:Alive() and (not v.LastHungerUpdate or CurTime() - v.LastHungerUpdate > 10) then
+		if v:Alive() and (not v.LastHungerUpdate or CurTime() - v.LastHungerUpdate > time) and !GAMEMODE.Config.hungerexcludes[v:Team()] then
 			v:hungerUpdate()
+		elseif GAMEMODE.Config.hungerexcludes[v:Team()] then
```
Comment (FPtje) [inline gamemode/modules/hungermod/sv_hungermod.lua]: Nonsensical config option. Why would you exlude teams of players from hunger? That's ridiculous. The select few who would have a good reason for this should DIY it.

File: gamemode/modules/hungermod/sv_hungermod.lua [hunk 5]
```diff
@@ -1,27 +1,34 @@
...
 hook.Add("PlayerSpawn", "HMPlayerSpawn", HMPlayerSpawn)
 
 local function HMThink()
-	if not GAMEMODE.Config.hungerspeed then return end
+	if !GAMEMODE.Config.hungerspeed then return end
+	local time = GAMEMODE.Config.hungertimer or 10
 
 	for k, v in pairs(player.GetAll()) do
-		if v:Alive() and (not v.LastHungerUpdate or CurTime() - v.LastHungerUpdate > 10) then
+		if v:Alive() and (not v.LastHungerUpdate or CurTime() - v.LastHungerUpdate > time) and !GAMEMODE.Config.hungerexcludes[v:Team()] then
```
Comment (FPtje) [inline gamemode/modules/hungermod/sv_hungermod.lua]: Negation style inconsistent with the rest of the code

File: gamemode/modules/hungermod/sv_hungermod.lua (hunk 4 above)
Comment (00875) [inline gamemode/modules/hungermod/sv_hungermod.lua]: https://github.com/FPtje/DarkRP/issues/2106 https://github.com/FPtje/DarkRP/issues/2106 https://github.com/FPtje/DarkRP/issues/2106
```
I ask for this as I have a few special jobs that are classed as "Out of RP", so I would like Hunger not to apply for those jobs.
```

File: entities/entities/food/init.lua (hunk 2 above)
Comment (00875) [inline entities/entities/food/init.lua]: I want more to do more accurate values.

Comment (FPtje) [inline entities/entities/food/init.lua]: That's bullshit.

File: gamemode/modules/hungermod/sv_hungermod.lua (hunk 4 above)
Comment (FPtje) [inline gamemode/modules/hungermod/sv_hungermod.lua]: This solution is way too specific for @Hazbelll's needs and wouldn't benefit **anyone** else

Comment (Hazbelll) [inline gamemode/modules/hungermod/sv_hungermod.lua]: I suppose this is true @FPtje although a few other communities I do dev work for also have their own select jobs that too would benefit from being excluded from hunger.

A much more global example would be Administrators on Duty. I imagine nearly every DarkRP server out there has a job for on-duty administration personnel too (I've yet to be part or even played on any that do not). They could benefit from this too.

Respect your choice either way. I'll try and sort something out if you do not wish for this in your repo.

File: gamemode/modules/hungermod/sv_hungermod.lua [hunk 6]
```diff
@@ -78,6 +85,8 @@ local function BuyFood(ply, args)
 
 		SpawnedFood.foodItem = v
 		SpawnedFood:Spawn()
+		
+		hook.Call("playerBoughtFood",nil,ply,v,SpawnedFood,cost)
```
Comment (FPtje) [inline gamemode/modules/hungermod/sv_hungermod.lua]: Do you *never* put spaces after commas?

Comment (00875) [inline gamemode/modules/hungermod/sv_hungermod.lua]: Yes. Why this is needed? I always write as:
```"DELETE FROM dstorage_pocket WHERE ID = '"..id.."' AND PLY = '"..ply:UniqueID().."';"``
```DStorage.Pocket.Remove(ply,id)```

**Comment by 00875** [comment]
No. I tired, i do endless coding about 2 weekends.

**Comment by FPtje** [comment]
I merged the playerBoughtFood commits

File: gamemode/modules/hungermod/sv_hungermod.lua (hunk 6 above)
Comment (FPtje) [inline gamemode/modules/hungermod/sv_hungermod.lua]: Because of god damn consistency with the rest of the code

**Pull request closed without merging**